### PR TITLE
Inkscape compatibility

### PR DESCRIPTION
**Pull request by kasbah**
As discussed in https://github.com/tracespace/pcb-stackup/issues/29. 

I am not sure what to do about escaping XML characters as the escape sequences, e.g. ` &quot;`, have semi-colons in them which need to be escaped in CSS and I am not sure if the double escaping will work at all.

EDIT: Looks like the `<[!CDATA[` is ok _without comment tags_. Works ok without comments in latest Chromium and Firefox as well. Do you know where it might cause problems to put it in without the comments?

### CODE REVIEW

File: test/stack-layers_test.js [hunk 1]
```diff
@@ -211,10 +211,11 @@ describe('stack layers function', function() {
         {tag: 'rect', attr: {x: -50, y: -50, width: 1100, height: 1100, fill: '#fff'}},
         {tag: 'use', attr: {'xlink:href': '#id_top_drl1'}},
         {tag: 'use', attr: {'xlink:href': '#id_top_drl2'}},
+        {tag: 'g', attr: {}, children: [0, 1, 2]},
         {
           tag: 'mask',
           attr: {id: 'id_top_mech-mask', fill: '#000', stroke: '#000'},
-          children: [0, 1, 2]
+          children: [3]
         }
```
Comment (kasbah) [inline test/stack-layers_test.js]: This fails, what am I doing wrong here?

**Comment by kasbah** [comment]
The last commit fails as I can't seem to adapt them to the grouping change. Not sure what I am doing wrong but the 'g' tag isn't detected by the tests.

**Comment by kasbah** [comment]
I tested it on IE11 and the [modified SVG](https://github.com/kasbah/pcb-stackup-inkscape-compat/blob/master/arduino-uno_top.svg) seems as good as the [un-modifed one](https://github.com/kasbah/pcb-stackup-inkscape-compat/blob/7bb01befd5ac10926a0ed6dcd682fbff63788d75/arduino-uno_top.svg) too. Seems to me not putting comments around the `!CDATA` declaration should be fine. 

@mcous, any help with grouping?

File: test/stack-layers_test.js [hunk 2]
```diff
@@ -211,10 +211,11 @@ describe('stack layers function', function() {
         {tag: 'rect', attr: {x: -50, y: -50, width: 1100, height: 1100, fill: '#fff'}},
         {tag: 'use', attr: {'xlink:href': '#id_top_drl1'}},
         {tag: 'use', attr: {'xlink:href': '#id_top_drl2'}},
+        {tag: 'g', attr: {}, children: [0, 1, 2]},
         {
           tag: 'mask',
           attr: {id: 'id_top_mech-mask', fill: '#000', stroke: '#000'},
-          children: [0, 1, 2]
+          children: [3]
```
Comment (mcous) [inline test/stack-layers_test.js]: Hey sorry for the delay; work's been crazy and I've managed to come down with the flu. Fun times.

Looks like you just need to change the line below to match the changes you made here:

``` js
expect(result.defs).to.contain(values[4])
```

**Comment by mcous** [comment]
So was doing a little bit of SVG research for a gerber-to-svg idea I've been tossing around, and stumbled upon this: https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/SVG_Guidelines#Avoid_the_use_of_CDATA_sections

Looks like we can ditch the CDATA entirely!

**Comment by kasbah** [comment]
This is done now, let me know what you think. I opted for using `element` direct to create the use tags instead of overloading `useLayer` any further. 

I ran the pcb-stackup integration tests with this and then converted this with Inkscape:

image

File: lib/stack-layers.js [hunk 3]
```diff
@@ -86,10 +88,11 @@ module.exports = function(element, id, side, layers, drills, outline, useOutline
     var cfMaskId = idPrefix + 'cf-mask'
     var cfMaskAttr = {id: cfMaskId, fill: '#fff', stroke: '#fff'}
     var cfMaskShape = smLayerId
-      ? [useLayer(element, smLayerId)]
+      ? [element('use', {'xlink:href': '#' + smLayerId, fill: '#fff', stroke: '#fff'})]
```
Comment (mcous) [inline lib/stack-layers.js]: Can you jog my memory with these fill and stroke attributes? Were they really necessary? It seems weird to me that specifying them in the parent (i.e. the mask node) wouldn't be enough. Although maybe it makes a difference that now a `<g>` is the parent.

Could you explore (here and below at line 108) keeping this as a `useLayer` call and placing the fill and stroke attributes on the `<g>` element?

**Review by mcous — CHANGES_REQUESTED** [review]

File: lib/stack-layers.js (hunk 3 above)
Comment (kasbah) [reply lib/stack-layers.js]: They are definitely necessary. Seems to work just on the group as well though https://github.com/kasbah/pcb-stackup-inkscape-compat/commit/0eb79042c8f77ec14729d266e69a69b6af351d8c. I'll change it to that when I next get a chance.

Comment (mcous) [reply lib/stack-layers.js]: Cool, I'm 👍 then

**Comment by kasbah** [comment]
Ok, all done then. Ran integration again:

image

Not sure, am I supposed to "dismiss" your review after I make the changes?

**Review by mcous — APPROVED** [review]

**Comment by kasbah** [comment]
Oh, I would have considered this a major since we are changing the output. Minor at the very least so we are not changing the output on someone who has tried to clamp their version to bug fixes only. Not that I think many people are using core directly.

**Comment by mcous** [comment]
Version numbers are cheap; happy to call it a major

**Comment by kasbah** [comment]
I looked over semver.org and did a google search but couldn't find anything about function return changes. It's all just semantics anyway. :laughing:

**Pull request merged**